### DISCUSSION TITLE
API Headers [May 08]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2056] [Discord Threads]
> 🙋‍♂️ **Ditmas**

**Comment by o1lab** [comment]
> 💁 **o1lab**  
#### Use one header : xc-auth or xc-token  
#  
> 🙋‍♂️ **Ditmas**  
#### yes that's something i still need to change, but the requests comes through.. it's just that my fields are empty  
#  
> 🙋‍♂️ **Ditmas**  
#### so there is something wrong with my JSON? but when i do the exact same request through thunderbold on vscode it works  
#  
> 🙋‍♂️ **Ditmas**  
#### it's really weird  
#  
> 💁 **o1lab**  
#### 🙂  
#  
> 💁 **o1lab**  
#### please review your code once more  
#  
> 💁 **o1lab**  
#### I see JSON.stringidy ? you are supposed to send an object  
#  
> 🙋‍♂️ **Ditmas**  
#### really? thats very weird because my thunderbolt does the same and that works fine..  
#  
> 🙋‍♂️ **Ditmas**  
#### what should there be in place of stringify perhaps?  
#  
> 🙋‍♂️ **Ditmas**  
#### thunder client*  
#  
> 💁 **o1lab**  
#### sorry, you will have to lookup axios request. Im not at desk.  
#  
> 🙋‍♂️ **Ditmas**  
#### it just accepts one parameter which is the configuration you give with it  
#  
> 🙋‍♂️ **Ditmas**  
#### url, header, body  
#  
> 🙋‍♂️ **Ditmas**  
#### method  
#  
> 🙋‍♂️ **Ditmas**  
#### it works fine like that i think there is just something miss with my json.object  
#  
> 🙋‍♂️ **Ditmas**  
#### while again this works fine in a postman or thunder client.. just like i send.. it's kind of weird  
#  
> 🙋‍♂️ **Ditmas**  
#### problem is solved, it seems like nocoDB/Swagger doesnt support axios very well  
#  
> 🙋‍♂️ **Ditmas**  
#### u need to use fetch  
#  
> 🙋‍♂️ **Ditmas**  
#### old school fetch  
#  
> 👨‍🚀 **wingkwong**  
#### > it seems like nocoDB/Swagger doesnt support axios very well   
####    
#### just had a quick test. it works fine using axios. the data can be inserted into DB and it can be shown in NocoDB UI.   
####    
#### ```   
#### let headersList = {   
####   "xc-auth": "<REDACTED>",   
####   "Content-Type": "application/json; charset=utf-8",   
####   Accept: "application/json",   
#### };   
####    
#### let bodyContent = {   
####   Title: "test",   
####   Title2: "test2",   
#### };   
####    
#### let reqOptions = {   
####   headers: headersList,   
####   body: bodyContent,   
#### };   
####    
#### const { data } = axios.post("<API_URL>", bodyContent, reqOptions);   
#### ```  
#  
> 👨‍🚀 **wingkwong**  
#### "xc-auth": "9bYhm5IA6taqBxFyey-K8nRrJeXMZ49JWonriVnu",   
####     "xc-token": "9bYhm5IA6taqBxFyey-K8nRrJeXMZ49JWonriVnu",   
####    
#### 1. like o1lab said, u just need either one.    
#### 2. just a reminder - the value in your xc-auth is API token, not Auth token. You may take a look at https://docs.nocodb.com/developer-resources/accessing-apis to see the differences n how to retrieve it  
#  
> 🙋‍♂️ **Ditmas**  
#### Thanks for the answord :)  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/970321352940736513)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
